resources/views/pages/user/[user]/game/[game]/compare.blade.php

```diff
@@ -67,12 +64,12 @@
         {!! gameAvatar($game->toArray(), label: false, iconSize: 48, iconClass: 'rounded-sm') !!}
         <h1 class="mt-[10px] w-full">Compare Unlocks</h1>
 
-        @if ($canModerate)
+        @can('viewSessionHistory', App\Models\PlayerGame::class)
             <x-hidden-controls-toggle-button>Moderate</x-hidden-controls-toggle-button>
-        @endif
+        @endcan
     </div>
 
-@if ($canModerate)
+@can('viewSessionHistory', App\Models\PlayerGame::class)
```

Comment:
Is there some reason to prefer calling `can` twice versus storing the result in a variable?

Reply:
Yes. It's worth noting that using `@can()` twice in this way does indeed call the policy function each time it's used in the template, but this is a measured choice and reflects a preference for reinforcing the declarative nature of the Blade template.

My opinion is that template languages should emphasize clarity and simplicity. By using `@can()` directly, the template makes it immediately clear to the reader what permissions are required for the relevant sections. The maintainer doesn't need to cross-check an imperative variable.

This is a weakly-held opinion though. Happy to revert if you prefer the `$canModerate` approach.

Reply:
I generally prefer to minimize function calls for performance reasons. What if the policy requires a DB query?

In this case it probably doesn't matter.